### PR TITLE
PERF-#4890: `PandasDataframeAxisPartition.drain` should be serialized only once

### LINKED ISSUE #4890: PERF: `drain` function should be serialized only once

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4890 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/core/dataframe/pandas/partitioning/axis_partition.py [hunk 1]
```diff
@@ -213,3 +213,22 @@ def deploy_func_between_two_axis_partitions(
         # for func, we fold args into kwargs. This is a bit of a hack, but it works.
         result = func(lt_frame, rt_frame, *kwargs.pop("args", ()), **kwargs)
         return split_result_of_axis_func_pandas(axis, num_splits, result)
+
+    @classmethod
+    def drain(cls, df, call_queue):
```
Comment (jbrockmendel) [inline modin/core/dataframe/pandas/partitioning/axis_partition.py:218]: annotations?

File: modin/core/execution/dask/implementations/pandas_on_dask/partitioning/virtual_partition.py [hunk 2]
```diff
@@ -418,14 +418,12 @@ def drain_call_queue(self, num_splits=None):
         num_splits : int, default: None
             The number of times to split the result object.
         """
-
-        def drain(df):
-            for func, args, kwargs in self.call_queue:
-                df = func(df, *args, **kwargs)
-            return df
-
+        # TODO: Need to check if `drain_call_queue` speeds up if helper
```
Comment (mvashishtha) [inline modin/core/execution/dask/implementations/pandas_on_dask/partitioning/virtual_partition.py:421]: Should we check this before merging the code?

Comment (anmyachev) [reply modin/core/execution/dask/implementations/pandas_on_dask/partitioning/virtual_partition.py:421]: I think it can be done separately.

**Review by mvashishtha — APPROVED** [review]

File: modin/core/execution/dask/implementations/pandas_on_dask/partitioning/virtual_partition.py (hunk 2 above)
Comment (YarShev) [reply modin/core/execution/dask/implementations/pandas_on_dask/partitioning/virtual_partition.py:421]: When do you want to remove this comment?

Comment (anmyachev) [reply modin/core/execution/dask/implementations/pandas_on_dask/partitioning/virtual_partition.py:421]: When someone who profiles on Dask sees the problem or lack of it.

**Review by YarShev — APPROVED** [review]

**Pull request merged**